### DISCUSSION TITLE
Why types of cards are supported?

### DISCUSSION THREAD

**Discussion by pazitka** [OpenSC/OpenSC#3038] [Q&A] [answered]
Hello.

Cards is some kind of secret knowledge and I can't even ask questions about it because I don't know what to ask...
I have Mifare 1k cards and I cant use them with OpenSC.
SIM cards are also not working and same with EMV.

So please tell me what should I buy to make it work and be able to write certificate to the card.
I don't care is it smartcard or NFC tag or something else just please tell me why it is not working.

**Comment by popovec** [comment, marked as answer]
OpenSC supports such cards/tokens that support some public key cryptographic operation. Whether it is an RSA or an EC signature. This type of operation is necessary to verify the identity of the device/user and is used, for example, for logging in to a remote computer, or browser verification for a WEB server, or creating an openvpn tunnel.

Mifare 1k/4k does not support EC or RSA signature/decryption. This is not supported in Mifare DESfire either.. According to the wiki (https://en.wikipedia.org/wiki/MIFARE) MIFARE SAM AV2 has RSA support, but I have never met this card and it is not supported in OpenSC - https://github.com/OpenSC/OpenSC/wiki/Supported-hardware-(smart-cards-and-USB-tokens) 

In the given link you have a list of supported tokens.. If you plan to store your certificate on the token, you need a non read only card, personalizable card. 

The list does not include the Yubico 5 token, which supports the PIV standard and is supported in OpenSC, so you can safely use this token as well.

Alternatively, if you feel like experimenting a bit, you can also create a token, which is supported in OpenSC, from cheap hardware - the STLINKv2 programmer (google -> "GNUK")
